instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/http/HttpClientAttributesExtractor.java

```diff
@@ -54,7 +54,49 @@ public static <REQUEST, RESPONSE> HttpClientAttributesExtractorBuilder<REQUEST,
   @Override
   public void onStart(AttributesBuilder attributes, Context parentContext, REQUEST request) {
     super.onStart(attributes, parentContext, request);
-    internalSet(attributes, SemanticAttributes.HTTP_URL, getter.url(request));
+    internalSet(attributes, SemanticAttributes.HTTP_URL, stripSensitiveData(getter.url(request)));
+  }
+
+  @Nullable
+  private static String stripSensitiveData(@Nullable String url) {
+    if (url == null || url.isEmpty()) {
+      return url;
+    }
+
+    int schemeEndIndex = url.indexOf(':');
+
+    if (schemeEndIndex == -1) {
+      // not a valid url
+      return null;
+    }
+
+    int len = url.length();
+    if (len <= schemeEndIndex + 2
+        || url.charAt(schemeEndIndex + 1) != '/'
+        || url.charAt(schemeEndIndex + 2) != '/') {
+      // has no authority component
+      return null;
+    }
+
+    // look for the end of the host:
+    //   ':' ==> start of port, or
+    //   '/', '?', '#' ==> start of path
+    int index;
+    for (index = schemeEndIndex + 3; index < len; index++) {
+      char c = url.charAt(index);
+      if (c == '/' || c == '?' || c == '#') {
+        break;
+      }
+    }
+
+    int atIndex = url.indexOf("@");
```

Comment:
then don't need to worry about query string
```suggestion
    int atIndex = url.lastIndexOf("@", index - 1);
```